src/testFixtures/java/org/opensearch/neuralsearch/BaseAgenticSearchRemoteModelIT.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import com.google.common.collect.ImmutableList;
+import org.opensearch.neuralsearch.query.AgenticSearchQueryBuilder;
+import org.opensearch.client.Response;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentType;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.opensearch.neuralsearch.util.RemoteModelTestUtils;
+
+import org.junit.After;
+import org.junit.Before;
+import java.io.IOException;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.message.BasicHeader;
+
+import lombok.SneakyThrows;
+import lombok.extern.log4j.Log4j2;
+
+import static org.opensearch.neuralsearch.util.TestUtils.DEFAULT_USER_AGENT;
+
+/**
+ * Base class for Agentic Search Remote Model integration tests.
+ */
+@Log4j2
+public abstract class BaseAgenticSearchRemoteModelIT extends BaseNeuralSearchIT {
+
+    protected static final String TEST_INDEX = "test-agentic-index";
+    protected static final String TEST_QUERY_TEXT = "Find documents about machine learning";
+
+    protected static String TEST_AGENT_ID;
+    protected static String TEST_MODEL_ID;
+    protected static String TEST_CONNECTOR_ID;
```

Comment:
having those variables as static can cause problems later when test are executed in parallel. Maybe good for this version, but consider refactoring them into class variables